### PR TITLE
Pipeline script to execute only the test cases which can be executed in IBM cloud

### PR DESCRIPTION
**Pull request by manasagowri**
Pipeline script to execute only the test cases which can be executed in IBM cloud

Signed-off-by: mgowri <mgowri@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: pipeline/special-case-executor.groovy [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            } else {
+                def updateContent = [
+                    "${tierLevel}": releaseContent[previousTierLevel]
+                ]
+                latestContent += updateContent
+            }
+
+            sharedLib.writeToReleaseFile(majorVersion, minorVersion, latestContent)
+            println "latest content is: ${latestContent}"
+        }
```
Comment (psathyan) [inline pipeline/special-case-executor.groovy:117]: This not being the primary execution space... let's avoid updating the recipe files

File: pipeline/special-case-executor.groovy [hunk 2]
```diff
@@ -0,0 +1,186 @@
+/*
+    Pipeline script for executing Tier x test suites for RH Ceph Storage.
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply pipeline/special-case-executor.groovy]: Done.

File: pipeline/special-case-executor.groovy [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+           Read the release yaml contents to get contents,
+           before other listener/Executor Jobs updates it.
+        */
+        releaseContent = sharedLib.readFromReleaseFile(
+            majorVersion, minorVersion, lockFlag=false
+        )
+
+        // Till the pipeline matures, using the build that has passed tier-0 suite.
+        testStages = sharedLib.fetchStages(cliArgs, tierLevel, testResults)
+        testStages = testStages.findAll { !it.key.contains("psi-only")}
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply pipeline/special-case-executor.groovy]: Done.

File: pipeline/special-case-executor.groovy [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        tierTypeStrings = previousTierLevel.split("-")
+        def tierLevelInt = tierTypeStrings[1].toInteger() + 1
+        tierLevel = tierTypeStrings[0] + "-" + tierLevelInt
+
+        // Once the pipeline matures, uncomment the below line
+        // cliArg = "--build ${tierLevel}"
+        def cliArgs = "--build tier-0"
+        if ( ciMap.artifact.build == "signed" ) {
+            cliArgs = "--build rc"
+        }
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply pipeline/special-case-executor.groovy]: Done.

File: pipeline/special-case-executor.groovy (hunk 1 above)
Comment (manasagowri) [reply pipeline/special-case-executor.groovy]: Done.

**Review by psathyan — APPROVED** [review]

**Pull request merged**